### PR TITLE
CLI docs section refresh

### LINKED ISSUE #1420: CLI section refresh
The [CLI docs](https://tskit.dev/msprime/docs/latest/cli.html) section needs a refresh and update. Some information is out of date.

@winni2k, you're the closest thing to a CLI expert here, would you be able to pick this one up at some point (not urgent)? 

The new [documentation docs](https://tskit.dev/msprime/docs/latest/development.html#documentation) should help orient you with the new way things are done.

### PR DESCRIPTION
**Pull request by winni2k**
This closes #1420 

- Adds the mutate command and redoes some of the wording in the msp introduction.
- Removes the alpha warning.

### CODE REVIEW

**Comment by winni2k** [comment]
@jeromekelleher was there anything else you were looking to see in the docs refresh?

**Comment by winni2k** [comment]
Can we move the CLI docs up one level out of the undescriptive "Utilities" section? It just seems wrong to have a utilities section in the docs. It's like saying "if you can't find what you are looking for in the API or Running Simulations section, expand this tab and have another go. I find myself expanding the utilities section as the first thing I do when I hit the docs page...

**Comment by jeromekelleher** [comment]
Definitely open to ideas on better structuring, and yes, "utilities" basically ended up being anything I couldn't figure out a better home for.

File: docs/cli.md [hunk 1]
```diff
@@ -20,19 +16,25 @@ should use the Python API where possible.
...
+simulation storing the results in a file. The `mutate` command places 
+mutations onto an existing tree sequence stored in an input file and writes
+the resulting tree sequence to an output file.
 
-:::{warning}
+:::{note}
 
-This tool is very new, and the interface may need to change
-over time. This should be considered an alpha feature!
+The way in which recombination and mutation rates are specified
```
Comment (jeromekelleher) [inline docs/cli.md:28]: I'm not sure this is helpful at this point, I think we can probably delete it.

File: docs/cli.md [hunk 2]
```diff
@@ -2,11 +2,7 @@
 
 # Command line interface
 
-```{eval-rst}
-.. todo:: This is out of date, needs a pass through to modernise.
-```
-
-Two command_line applications are provided with `msprime`: {ref}`sec_msp` and
+Two command line applications are provided with `msprime`: {ref}`sec_msp` and
 {ref}`sec_mspms`. The {command}`msp` program is an experimental interface for
```
Comment (jeromekelleher) [inline docs/cli.md:6]: Probably can't call it "experimental" any more.

**Review by jeromekelleher — COMMENTED** [review]
LGTM @winni2k, thanks.

I suppose the confusing thing at this point is that ``msp simulate`` is an interface to the deprecated ``simulate()`` function, whereas ``msp mutate`` is for the new ``sim_mutations`` function. These will differ in that they have different defaults for discrete/continuous genome, etc.

The pass through the docs here is really to get some fresh eyes on them, to see if they are actually any use. What is ``msp simulate`` for? 

On the other hand, I can see ``msp mutate`` being quite useful as a way of throwning mutations down on tree sequences from the shell.

**Comment by winni2k** [comment]
> The pass through the docs here is really to get some fresh eyes on them, to see if they are actually any use. What is msp simulate for?

Yeah, msp simulate does more than it needs to. Is it too late to deprecate `msp simulate` in favor of using `msp ancestry`? Or can we remove the mutation rate argument and set it to zero? And then maybe rename msp simulate to `msp ancestry`? Is this our final chance to make radical changes to the CLI, or is it already too late for that?

**Comment by winni2k** [comment]
Reworked the msp sub-command descriptions to make it clearer to potential users what these sub-commands are used for. I assume `msp ancestry` will soon be coming to an msp CLI near you?

**Comment by jeromekelleher** [comment]
Yes, this is great, thanks @winni2k! I think you're right, ``msp ancestry`` and ``msp mutate`` are the right commands.  I don't have any immediate plans for writing ``msp ancestry``, but I guess we could make a very simple minimal version for 1.0?

**Comment by winni2k** [comment]
Sure!

**Comment by winni2k** [comment]
Here's a quick and dirty implementation of `msp ancestry`. I have renamed `msp simulate` to `msp ancestry`. `msprime.sim_ancestry` is used on the backend, the mutation rate gets ignored with a warning if set above 0, and `msp simulate` is an alias for `msp ancestry`. 

I can't figure out why the `ancestry` subcommand can't be found by the sphinx parser 🤷

**Comment by winni2k** [comment]
Do we want to retain the option names from `msp simulate` for `msp ancestry`?

**Comment by jeromekelleher** [comment]
I think we should keep the ``simulate`` command as-is for compatibility, and make a completely new ``ancestry`` command which supports a (small) subset of the options for ``sim_ancestry``.

Arguably, we should rename ``mutate`` to ``mutations`` so that we have a straightforward relationship between the commands and the functions, but I can't remember now if we've released a ``msp mutate`` command or not already.

**Comment by winni2k** [comment]
v0.7.5 has no mutate command yet, but there is a mutations command ("Dump mutations in tabular format.").

**Comment by winni2k** [comment]
I have redone the previous commit. `msp mutate` now raises a deprecation warning when used. `msp ancestry` is a separate command. New docs are also included.

File: msprime/cli.py [hunk 3]
```diff
@@ -1115,6 +1134,47 @@ def add_simulate_subcommand(subparsers) -> None:
...
+        "-N",
+        type=float,
+        default=1,
+        help="The number of individuals in the population",
+    )
+    parser.add_argument(
+        "--ploidy",
+        "-k",
+        type=int,
+        default=1,
```
Comment (jeromekelleher) [inline msprime/cli.py:1171]: Probably should default to 2, for consistency with sim_ancestry.

**Review by jeromekelleher — COMMENTED** [review]
Looks great, thanks @winni2k! OK, just one minor thing and we're ready to squash and merge I think.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**